data/vm/vmOutputApi.go

```diff
@@ -0,0 +1,72 @@
+package vm
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// VMOutputApi is a wrapper over the vmcommon's VMOutput
+type VMOutputApi struct {
+	ReturnData      [][]byte                     `json:"returnData"`
+	ReturnCode      string                       `json:"returnCode"`
+	ReturnMessage   string                       `json:"returnMessage"`
+	GasRemaining    uint64                       `json:"gasRemaining"`
+	GasRefund       *big.Int                     `json:"gasRefund"`
+	OutputAccounts  map[string]*OutputAccountApi `json:"outputAccounts"`
+	DeletedAccounts [][]byte                     `json:"deletedAccounts"`
+	TouchedAccounts [][]byte                     `json:"touchedAccounts"`
+	Logs            []*LogEntryApi               `json:"logs"`
+}
+
+// StorageUpdateApi is a wrapper over vmcommon's StorageUpdate
+type StorageUpdateApi struct {
+	Offset []byte `json:"offset"`
+	Data   []byte `json:"data"`
+}
+
+// OutputAccountApi is a wrapper over vmcommon's OutputAccount
+type OutputAccountApi struct {
+	Address        string                       `json:"address"`
+	Nonce          uint64                       `json:"nonce"`
+	Balance        *big.Int                     `json:"balance"`
+	BalanceDelta   *big.Int                     `json:"balanceDelta"`
+	StorageUpdates map[string]*StorageUpdateApi `json:"storageUpdates"`
+	Code           []byte                       `json:"code"`
+	CodeMetadata   []byte                       `json:"codeMetaData"`
+	Data           []byte                       `json:"data"`
+	GasLimit       uint64                       `json:"gasLimit"`
+	CallType       vmcommon.CallType            `json:"callType"`
+}
+
+// LogEntryApi is a wrapper over vmcommon's LogEntry
+type LogEntryApi struct {
+	Identifier []byte   `json:"identifier"`
+	Address    string   `json:"address"`
+	Topics     [][]byte `json:"topics"`
+	Data       []byte   `json:"data"`
+}
+
+// GetFirstReturnData is a helper function that returns the first ReturnData of VMOutput, interpreted as specified.
+func (vmOutput *VMOutputApi) GetFirstReturnData(asType vmcommon.ReturnDataKind) (interface{}, error) {
```

Comment:
unit test?

Reply:
added unit tests for all cases (as big int, as big int string, as hex and as string)